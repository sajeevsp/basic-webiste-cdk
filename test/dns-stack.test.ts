import { SynthUtils } from "@aws-cdk/assert";
import * as cdk from "@aws-cdk/core";
import DNSStack from "../lib/dns-stack";

test("DNS Stack", () => {
  const app = new cdk.App();
  // WHEN
  const siteDomain = "example.com";
  const siteSubDomain = "www.example.come";
  const environments = {
    "us-east": {
      env: {
        account: "000000000000",
        region: "us-east-1",
      },
    },
  };

  //CDK stack - self deploys itself and then deploys the site stack.
  const stack = new DNSStack(
    app,
    "DNSStack",
    {
      siteDomain: siteDomain,
      siteSubDomain: siteSubDomain,
    },
    environments["us-east"]
  );
  // THEN
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
