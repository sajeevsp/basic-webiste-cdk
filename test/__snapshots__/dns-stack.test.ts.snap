// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DNS Stack 1`] = `
Object {
  "Resources": Object {
    "DomainARecordC223EFD6": Object {
      "Properties": Object {
        "AliasTarget": Object {
          "DNSName": "s3-website-us-east-1.amazonaws.com",
          "HostedZoneId": "Z3AQBSTGFYJSTF",
        },
        "HostedZoneId": "DUMMY",
        "Name": "example.com.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "SubDomainARecord1F984879": Object {
      "Properties": Object {
        "AliasTarget": Object {
          "DNSName": "s3-website-us-east-1.amazonaws.com",
          "HostedZoneId": "Z3AQBSTGFYJSTF",
        },
        "HostedZoneId": "DUMMY",
        "Name": "www.example.come.example.com.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
  },
}
`;
